### ISSUE TITLE
[bug report] F# project was not excluded from Accord.NET.sln by default.

### ISSUE THREAD

**Issue by Chinese-Xu** [accord-net/framework#916]
F# required projects:
Accrod.Tests.MachineLearning.FSharp (Excluded by default)
Accrod.Tests.Math.FSharp

Build Accord.NET.sln failed when F# was not installed in Visual Studio 2015

Thanks,
Wiley Xu

**Comment by cesarsouza** [comment]
Hi Wiley,

Thanks a lot! I will update the documentation with more details about the F# projects, probably stating that they are actually example/unit tests and can be disabled/unloaded by the user when they open the solution in case they do not have F# installed.

Thanks again,
Cesar

**Comment by Chinese-Xu** [comment]
I suggest that we should exclude all FSharp projects by default.   And use will notice that 5 projects be skipped when build Accrod.NET.sln

**Comment by cesarsouza** [comment]
@Chinese-Xu I agree it would be best to minimize the number of possible problems the user can find when first opening the project.

However, what happens when you open the project in VS2015 when F# support is not installed? Does it fail to open completely, or simply doesn't build?

If it only fails to build, maybe I could disable them from the Debug configuration and include them only in the net462 (release) configuration. Or maybe create a special configuration option only for building FSharp projects, but I would rather avoid that if possible.

Regards,
Cesar

**Comment by dsyme** [comment]
> However, what happens when you open the project in VS2015 when F# support is not installed? Does it fail to open completely, or simply doesn't build?

Just require that F# support is installed in order to develop.  It's not hard to add it.  Marginalizing F# support just because people can't install pre-requisites is not a good path

**Comment by cesarsouza** [comment]
Hi @dsyme,

You are right. Installing F# shouldn't be that much hard anyways (but I think it wasn't being included by default in VS2015, right?) Anyways, thanks a lot for the input.

@Chinese-Xu, do you think you could try to add F# to your VS2015 installation and please share with us how you did that? This way I could include better documentation that actually documents how this needs to be done.

Regards,
Cesar

**Comment by Chinese-Xu** [comment]
Hi @cesarsouza , I will attach related steps of enable F# in VS at 8 oct.

**Comment by Chinese-Xu** [comment]
Hi @cesarsouza , the main steps to install F# into VS2015 is

1. Open "Programs and Features" form
2. Select "Microsoft Visual Studio Community 2015 Updates" from program list, or any other version have been  installed.
3. Click "Change" button above program list to trigger the maintain dialog for VS 2015.
4. Wait until the initial process done
5. Click "Modify" button on the bottom of dialog
6. Check "Program Language" -> "Visual F#" from component list, than click "Next" button.
7. Waiting all process finished.

Below are the screen shorts for above steps.  But the issue is the UI language of my VS is Chinese,  I will try to recapture those images on en-us environment in recent days.

![01_open programm features](https://user-images.githubusercontent.com/2484031/31307674-a2f6a8ec-ab9b-11e7-873e-5f87c317f2b9.png)
![02_waiting for next step](https://user-images.githubusercontent.com/2484031/31307677-ab283ce2-ab9b-11e7-8beb-5c003a3c322c.png)
![03_click modify](https://user-images.githubusercontent.com/2484031/31307678-ad9f3a48-ab9b-11e7-9081-9331a4564a70.png)
![04_check f](https://user-images.githubusercontent.com/2484031/31307681-b5e95846-ab9b-11e7-82a6-868214e79e87.png)
![05_confirm](https://user-images.githubusercontent.com/2484031/31307682-bba7eaf4-ab9b-11e7-8b34-fa23f4302943.png)
![06_installing](https://user-images.githubusercontent.com/2484031/31307683-c22622ba-ab9b-11e7-8497-a117a4fe57a1.png)

**Comment by cesarsouza** [comment]
Hi @Chinese-Xu,

Thanks a lot! I wasn't sure if it was needed to download some external pack (i.e. like [Visual F# Tools](https://www.microsoft.com/en-us/download/details.aspx?id=48179&751be11f-ede8-5a0c-058c-2ee190a24fa6=True)) separately. But now from your answer I assume that everything can be get from the VS installer. I will update the readme with the instructions you provided, thanks again!

Regards,
Cesar
